ios/FluentUI/BottomSheet/BottomSheetViewController.swift

```diff
@@ -0,0 +1,404 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+enum BottomSheetViewState {
+    case expand
+    case collapse
+}
+
+@objc(MSFBottomSheetViewControllerDelegate)
+public protocol BottomSheetViewControllerDelegate: AnyObject {
+    /// Called when drawer is being expanded.
+    @objc optional func bottomSheetViewControllerDidExpand(_ controller: BottomSheetViewController)
+
+    /// Called when drawer is being collapsed.
+    @objc optional func bottomSheetViewControllerDidCollapse(_ controller: BottomSheetViewController)
+}
+
+/// BottomSheetViewController is a module component that should be persistently shown on the screen and hosting various components relevant to on-screen content.
+/// On iPhone devices, the BottomSheetViewController is meant to be thumb friendly position at the bottom of the screen which users can pan or tap on resize handle to expand and collapse while still be able to interact with other UIControls outside of this module component. On iPad regular horizontal size, the BottomSheetViewcontroller is shown as floating UIView anchored on the bottom of the screen and not expandable. In order to make this persistent view work with various iOS assistive technology, instead of presenting this viewcontroller, initialize the viewcontroller and add its view directly as subview of its hosting viewcontroller.
+@objc(MSFBottomSheetViewController)
+public class BottomSheetViewController: UIViewController {
+    @objc public weak var delegate: BottomSheetViewControllerDelegate?
+    /// BottomSheetViewController's view height in collapsed mode.
+    @objc public var collapsedHeight: CGFloat = Constants.collapsedHeight {
+        didSet {
+            updateFrame()
+        }
+    }
+
+    /// BottomSheetViewController's view height in expanded mode. If the value is -1, it will grow to the its windows safe area height.
+    @objc public var preferredExpandedHeight: CGFloat = -1 {
+        didSet {
+            updateFrame()
+        }
+    }
+
+    /// BottomSheetViewController's view width in iPad when horizontal size is regular
+    @objc public var preferredContentWidthForPad: CGFloat = Constants.contentWidthForPad {
+        didSet {
+            updateFrame()
+        }
+    }
+
+    @objc public var showResizeHandle: Bool = true {
```

Comment:
Should this be public? Do we ever want an expandable sheet without the handle or vice versa? It could just be an internal side effect of whether the sheet is expandable or not.

Reply:
i think i saw one possible example in Word app spec that bottom sheet is locked and not expandable.

Reply:
I know, I mean it might be more natural to expose that through an "isExpandable" flag, rather than referencing the resize handle. The handle is more of a UI detail of how we represent expandability.